plugins/destination/postgresql/client/insert.go

```diff
@@ -43,10 +43,10 @@ func (c *Client) InsertBatch(ctx context.Context, messages message.WriteInserts)
 		if !ok {
 			return fmt.Errorf("table name not found in metadata")
 		}
-		table := tables.Get(tableName)
-		if table == nil {
+		if _, ok = c.pgTablesToPKConstraints[tableName]; !ok {
```

Comment:
if we store the table with `c.pgTablesToPKConstraints[tableName] = ""` we need to get the value and check it for empty too (or just that since it's a string)
```suggestion
		if c.pgTablesToPKConstraints[tableName] == "" {
```

Reply:
We should never store such table, actually (pg always names constraints automatically, if no name is provided).
So we either query & get the proper name or we store during table creation & still have non-empty data.

Reply:
We should never store such table, actually (pg always names constraints automatically, if no name is provided).
So we either query & get the proper name or we store during table creation & still have non-empty data.

Reply:
When we "store" `""` as a value we basically say that we actually saw the table.
However, I see that it might be confusing, so, knowing that the constraint will always have a name in PG, I'll just remove the confusing `switch` from `list_tables.go`

Reply:
Simplified code in 254ce2abd18ba9dd0fabebff89d30c90f95f9a91

Reply:
Had to move back in 08cc7bc476131bcd64f68c24c799c87ba0a21af0 (added comment why we do that)